allennlp/data/dataset_readers/__init__.py

```diff
@@ -1,10 +1,14 @@
+from typing import Dict, Type  # pylint: disable=unused-import
+
 from .dataset_reader import DatasetReader
 from .language_modeling import LanguageModelingReader
 from .snli import SnliReader
 from .squad import SquadSentenceSelectionReader
 from .sequence_tagging import SequenceTaggingDatasetReader
 
-dataset_readers = {}  # pylint: disable=invalid-name
+# pylint: disable=invalid-name
+dataset_readers = {}  # type: Dict[str, Type[DatasetReader]]
```

Comment:
Glad you figured this out, thanks.

Also, you can just put these type annotations directly in the code in python 3.6 (`dataset_readers: Dict[str, Type[DatasetReader]] = {}`).  We should probably see if we can switch now (not in this PR, though).